komodo/yaml_file_types.py

```diff
@@ -38,16 +133,521 @@ class ManifestFile(YamlFile):
+            )
+        return _komodo_error(
+            package=package,
+            version=version,
+            maintainer=repository_entries[package][version]["maintainer"],
+        )
+
+    def validate_repository_file(self) -> None:
+        repository_file_content: dict = self.content
+        message = (
+            "The file you provided does not appear to be a repository file "
+            "produced by komodo. It may be a release file. Repository files "
+            "have a format like the following:\n\n"
+            "pytest-runner:\n  6.0.0:\n    make: pip\n    "
+            "maintainer: scout\n    depends:\n      - wheel\n      - "
+            """setuptools\n      - python\n\npython:\n  "3.8":\n    ..."""
+        )
+        assert isinstance(repository_file_content, dict), message
+        errors = []
+        for package_name, versions in repository_file_content.items():
+            try:
+                Package.validate_package_name(package_name)
+                if not isinstance(versions, dict):
+                    errors.append(
+                        f"Versions of package '{package_name}' is not formatted"
+                        f" correctly ({versions})"
+                    )
+                    continue
+                validation_errors = self.validate_versions(package_name, versions)
+                if validation_errors:
+                    errors.extend(validation_errors)
+            except (ValueError, TypeError) as e:
+                errors.append(str(e))
+
+        handle_validation_errors(errors, message)
+
+    def validate_versions(self, package_name: str, versions: dict) -> List[str]:
+        errors = []
+        for version, version_metadata in versions.items():
+            Package.validate_package_version(package_name, version)
+            make_errors = Package.validate_package_make_with_errors(
+                package_name, version, version_metadata.get("make")
+            )
+            errors.extend(make_errors)
+            maintainer_errors = Package.validate_package_maintainer_with_errors(
+                package_name,
+                version,
+                version_metadata.get("maintainer"),
+            )
+            errors.extend(maintainer_errors)
+            for (
+                package_property,
+                package_property_value,
+            ) in version_metadata.items():
+                validation_errors = self.validate_package_properties(
+                    package_name,
+                    version,
+                    package_property,
+                    package_property_value,
+                )
+                errors.extend(validation_errors)
+        return errors
+
+    def validate_package_properties(
+        self,
+        package_name: str,
+        package_version: str,
+        package_property: str,
+        package_property_value: str,
+    ) -> List[str]:
+        pre_checked_properties = ["make", "maintainer"]
```

Comment:
Maybe a docstring is warranted here stating that what is returned is a list of errors.

Reply:
(and similarly for other new functions). It is not readable from the typing or the function name whether exceptions are thrown or just a list of errors are returned. When neither of these explain this, there should be a docstring stating it.

Reply:
Agreed!